### PR TITLE
feat(builderops): implement explicit promotion gateway (#1504)

### PR DESCRIPTION
**Pull request by RasmusTho**
Closes #1504
Parent: #1498
Depends on: #1499 / ADR-0010, #1502, #1503

## Summary
- Add a proposal-only BuilderOps PromotionIntent gateway for GitHub Issue drafts, PR/branch proposals, ADR/doc writeback proposals, generated projection proposals, and discard receipts.
- Add leased PromotionIntent transitions for accepted, promoted, rejected, and discarded, with idempotent BuilderOpsReceipt records carrying source_refs, proposal payloads, rationales, and result_refs where applicable.
- Add CLI commands for promotion preview, dry-run receipt creation, and explicit transition decisions.
- Document the gateway boundary and keep API/MCP autonomous execution separate from promotion execution.

## Changed files
- app/builderops/promotion_gateway.py
- app/builderops/store.py
- app/cli/builderops.py
- app/builderops/__init__.py
- docs/builderops/BUILDEROPS_PROMOTION_GATEWAY.md
- docs/builderops/BUILDEROPS_VAULT_STORE.md
- docs/builderops/BUILDEROPS_VAULT_BOUNDARY.md
- docs/builderops/BUILDEROPS_VAULT_OBJECT_MODEL.md
- docs/DOCS_INDEX.md
- tests/builderops/test_builderops_promotion_gateway.py
- tests/cli/test_builderops_cli.py

## Acceptance criteria mapping
- AC1: PromotionIntent transitions cover accepted, promoted, rejected, discarded; tests assert valid transitions, duplicate idempotent retry, and invalid terminal/pre-acceptance transitions.
- AC2: GitHub Issue dry-run preserves source_refs and appends a traceable receipt; tests assert draft body contains Parent, Source Anchors, and BuilderOps source_refs.
- AC3: Repo-doc/ADR promotion remains proposal-only; gateway renders no-write proposals and docs/tests assert normal PR workflow is required for authority-bearing repo changes.
- AC4: Discard transition creates a BuilderOpsReceipt with rationale and traceable source_refs.

## Validation
- PYTEST_DISABLE_PLUGIN_AUTOLOAD=1 /opt/homebrew/bin/python3.12 -m pytest -q tests -k builderops
  - 28 passed, 1 skipped, 4487 deselected, 69 warnings in 11.08s
  - Warnings are pre-existing unknown pytest.mark.asyncio warnings outside BuilderOps tests.
- /tmp/agentic-pkm-ruff-venv/bin/python -m ruff check app tests
  - All checks passed!
- git diff --check app tests docs
  - passed
- /opt/homebrew/bin/python3.12 scripts/docs_guard.py
  - Docs guard: OK
- /opt/homebrew/bin/python3.12 -m app.cli builderops --help
  - rendered successfully with promotion-preview, promotion-dry-run, and promotion-transition commands
- rg -n "PromotionIntent|discard receipt|source_refs" app tests docs
  - confirmed gateway, docs, and tests include required PromotionIntent, discard receipt, and source_refs coverage

## Dispatcher fallback
- python3 -m app.dispatcher status --json failed locally with ModuleNotFoundError: No module named 'yaml'.
- /opt/homebrew/bin/python3.12 -m app.dispatcher status --json ran, but reported db_exists:false.
- Used documented fallback: scripts/issue_pickup_claim.sh --issue 1504, then set the GitHub Project item to In Progress via GraphQL.

## Scope statement
This PR implements explicit BuilderOps promotion proposal/receipt/state-transition mechanics only. It does not create GitHub Issues, open PR branches, mutate repo docs/ADRs/skills, write generated projections, expose autonomous API/MCP gateway execution, or change product/runtime authority surfaces.

## Known follow-ups
- #1505 should generate repo projections from BuilderOps Vault after this gateway lands.
- The dispatcher yaml/bootstrap issue remains out of scope for #1504.

### CODE REVIEW

File: app/builderops/promotion_gateway.py [hunk 1]
```diff
@@ -0,0 +1,382 @@
...
+            )
+        proposal = self.render_proposal(intent_id)
+        refs = source_refs if source_refs is not None else intent["source_refs"]
+        receipt_extra: dict[str, Any] = {
+            "promotion_decision": normalized_decision,
+            "promotion_rationale": rationale,
+            "promotion_proposal": proposal,
+        }
+        if result_refs:
+            receipt_extra["result_refs"] = result_refs
```
[the remark this answers is not in the document]
Comment (RasmusTho) [reply app/builderops/promotion_gateway.py:189]: Daily review-comment closure audit repair: addressed in PR #1539 (commit cc29fb8d). Leaving this original thread unresolved until the repair PR merges.

File: app/builderops/promotion_gateway.py [hunk 2]
```diff
@@ -0,0 +1,382 @@
...
+    ) -> dict[str, Any]:
+        intent = self._require_intent(intent_id)
+        proposal = self.render_proposal(intent_id)
+        actor_ref = normalize_actor(actor)
+        refs = source_refs if source_refs is not None else intent["source_refs"]
+        receipt = self._store.append_receipt(
+            summary=f"Dry-run promotion proposal for {intent['id']}",
+            event_type="promotion_dry_run",
+            actor=actor_ref,
+            occurred_at=intent["updated_at"],
```
[the remark this answers is not in the document]
Comment (RasmusTho) [reply app/builderops/promotion_gateway.py:132]: Daily review-comment closure audit repair: addressed in PR #1539 (commit cc29fb8d). Leaving this original thread unresolved until the repair PR merges.

File: app/builderops/promotion_gateway.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (RasmusTho) [reply app/builderops/promotion_gateway.py:189]: Daily review-comment closure audit follow-up: repair PR #1539 has merged (merge commit 5bac6d3), so this original review finding is now addressed by merged code.

File: app/builderops/promotion_gateway.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (RasmusTho) [reply app/builderops/promotion_gateway.py:132]: Daily review-comment closure audit follow-up: repair PR #1539 has merged (merge commit 5bac6d3), so this original review finding is now addressed by merged code.
